tools/mason/MasonInit.chpl

```diff
@@ -0,0 +1,174 @@
+        const tomlFile = new owned(parseToml(toParse));
+        if tomlFile["brick"]["name"] {
+          projectName = tomlFile["brick"]["name"].s;
+        }
+        else {
+          throw new owned MasonError("Project Name doesn't exist in Mason.toml");
+        }
+        if tomlFile["brick"]["version"] {
+          version = tomlFile["brick"]["version"].s;
+        } 
+        else {
+          throw new owned MasonError("Version doesn't exist in Mason.toml");
+        }
+        validateAndInit(projectName, vcs=true, show=false);
+        checkVersion(version);
+      } 
+      //If Mason.toml not present, proceed with init
+      const cwd  = getEnv("PWD");
+      const currDir = basename(cwd);
+      const name = currDir;
+      const path = '.';
+      makeBasicToml(name, path);
+      ValidateInit(path);
+      writeln("Initialized new library project: " + name);
+    } 
+    else {
+      // if the target directory in path doesnt exist, throw error
+      // if target directory exists, check for files && validate
+      // create folders and toml file without overwriting anything 
+      // if TOML file exists, check for values in it and validate
+      const path = name;
+      if isDir(path) {
+        ValidateInit(path);
+        writeln("Initialized new library project in " + path + ": " + basename(path));
+      } 
+      else {
+        throw new owned MasonError("Directory does not exist:" + path +
+                              ". Did you mean 'mason new' to create a new project from scratch?"); 
+      }
+    }
+  }
+  catch e: MasonError {
+    writeln(e.message());
+    exit(1);
+  }
+}
+
+proc ValidateInit(path: string) throws {
+  var files = [ "/Mason.toml" , "/src" , "/test" , "/example", "/.git", ".gitignore" ];
+  var toBeCreated : list(string);
+  for idx in 1..files.size do {
+    const file = files(idx);
+    if file == "/Mason.toml" {
+      if isFile(path + file) {
+        //scan using TOML reader and validate
+        const toParse = open(path + file, iomode.r);
+        const tomlFile = new owned(parseToml(toParse));
+        var projectName = tomlFile["brick"]["name"].s;
+        var version = tomlFile["brick"]["version"].s;
+        validateAndInit(projectName, vcs=true, show=false);
+        checkVersion(version);
+      } 
+      else {
+        toBeCreated.append(file);
+      }
+    } 
+    else {
+      const dir = file;
+      if isDir(path + dir) == false {
+        toBeCreated.append(dir);
+      }
+    }
+  }
+  
+  if toBeCreated.size == 1 {
+    var fileName = "";
+    if path != '.' {
+      fileName = basename(path); 
+    } else {
+      const pwd = getEnv("PWD");
+      fileName = basename(pwd);
+    }
+    if isFile(path + '/src/' + fileName + '.chpl') == false {
+      makeModule(path,fileName);
+    } 
+    else {
+      throw new owned MasonError("Library project has already been initialised.");
+    } 
+  }
+
+  for file in toBeCreated{
```

Comment:
`file` is a type in the standard lib ([link](https://chapel-lang.org/docs/master/modules/standard/IO.html?highlight=file#IO.file)) - to avoid confusion, I'd suggest renaming, e.g. `masonFile`, `metafile`, `f`